internal/server/auditsink/logfile/logfile.go

```diff
@@ -0,0 +1,57 @@
+package logfile
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+
+	"go.flipt.io/flipt/internal/server/auditsink"
+	"go.uber.org/zap"
+)
+
+const sinkType = "logfile"
+
+// LogFileSink is the structure in charge of sending Audits to a specified file location.
+type LogFileSink struct {
+	logger *zap.Logger
+	file   *os.File
+	mtx    sync.Mutex
+}
+
+// NewLogFileSink is the constructor for a LogFileSink.
+func NewLogFileSink(logger *zap.Logger, filePath string) (*LogFileSink, error) {
+	file, err := os.OpenFile(filePath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
+	if err != nil {
+		return nil, err
+	}
+
+	lfs := &LogFileSink{
+		logger: logger,
+		file:   file,
+	}
+
+	return lfs, nil
+}
+
+func (l *LogFileSink) SendAudits(auditEvents []auditsink.AuditEvent) error {
+	l.logger.Debug("performing batched sending of audits", zap.Stringer("sink", l))
+	l.mtx.Lock()
+	defer l.mtx.Unlock()
+	for _, ae := range auditEvents {
+		b, err := json.Marshal(ae)
+		if err != nil {
+			l.logger.Error("failed to encode audit event", zap.Stringer("sink", l))
+			continue
+		}
+		_, err = l.file.Write(append(b, '\n'))
+		if err != nil {
+			l.logger.Error("failed to write audit event to file", zap.String("file", l.file.Name()))
+			continue
+		}
+	}
+	return nil
+}
```

Comment:
This could be slightly terser if you used a https://pkg.go.dev/encoding/json#Encoder

`encoder.Encode(ae)` will actually append a newline for you.

```suggestion
// NewLogFileSink is the constructor for a LogFileSink.
func NewLogFileSink(logger *zap.Logger, filePath string) (*LogFileSink, error) {
	file, err := os.OpenFile(filePath, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
	if err != nil {
		return nil, err
	}

	lfs := &LogFileSink{
		logger: logger,
		file:   file,
		enc: json.NewEncoder(file),
	}

	return lfs, nil
}

func (l *LogFileSink) SendAudits(auditEvents []auditsink.AuditEvent) error {
	l.logger.Debug("performing batched sending of audits", zap.Stringer("sink", l))
	
	l.mtx.Lock()
	defer l.mtx.Unlock()
	for _, ae := range auditEvents {
		_, err = l.enc.Encode(ae)
		if err != nil {
			l.logger.Error(
				"failed to write audit event to file",
				zap.String("file", l.file.Name()),
				zap.Error(err),
			)
			continue
		}
	}
	
	return nil
}
```

Reply:
Additionaly, I would put a `Close()` method on your `*LogFileSink` type which calls `file.Close()` (this is why I preserved the file too).

Otherwise, the bytes may not get flushed to disk before exit.

Reply:
Last though: are we OK with `continue` on error?
Maybe we should slirp up a multi-error and return it.
Otherwise, the caller might just poor stuff into this method and it always return nil.

Reply:
That is exactly what I was looking for. Some way to represent multiple errors in a readable format. Thanks George 🙏 big help.